libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/audio/AudioTypeResolver.kt

```diff
@@ -0,0 +1,86 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation.metrics.audio
+
+import android.content.Context
+import android.media.AudioManager
+import android.os.Build
+import com.mapbox.services.android.navigation.v5.internal.exception.NavigationException
+
+sealed class AudioTypeResolver {
+
+    companion object {
+        const val BLUETOOTH = "bluetooth"
+        const val HEADPHONES = "headphones"
+        const val SPEAKER = "speaker"
+        const val UNKNOWN = "unknown"
+    }
+
+    var chain: AudioTypeResolver? = null
+
+    open fun nextChain(chain: AudioTypeResolver) {
+        this.chain = chain
+    }
+
+    abstract fun obtainAudioType(context: Context): String
+}
+
+class BluetoothAudioType : AudioTypeResolver() {
```

Comment:
```suggestion
class Bluetooth : AudioTypeResolver() {
```
here and bottom. Sealed class is called as `AudioTypeResolver.Bluetooth(...)`